Makefile

```diff
@@ -23,22 +23,20 @@ ifeq ($(SKIP_GO_VERSION_CHECK),)
     include golang.mk
 endif
 
-ifneq ($(GOPATH),)
-    GOARCH=$(shell go env GOARCH)
-    ifeq ($(ARCH),)
-        ARCH = $(GOARCH)
-    endif
-
-    ARCH_DIR = arch
-    ARCH_FILE_SUFFIX = -options.mk
-    ARCH_FILE = $(ARCH_DIR)/$(ARCH)$(ARCH_FILE_SUFFIX)
-    ARCH_FILES = $(wildcard arch/*$(ARCH_FILE_SUFFIX))
-    ALL_ARCHES = $(patsubst $(ARCH_DIR)/%$(ARCH_FILE_SUFFIX),%,$(ARCH_FILES))
-
-    # Load architecture-dependent settings
-    include $(ARCH_FILE)
+GOARCH=$(shell go env GOARCH)
+ifeq ($(ARCH),)
```

Comment:
It might be worth adding a comment here explaining this is to allow the user to set `ARCH` before we overwrite it.

# More info

A quick look makes it appear as though this is wrong since `ARCH` isn't set yet. However, since `ARCH` is added to the `USER_VARS` variable, that check is to allow the user to define the architecture they want to build with (cross build). If not specified, we default to the host architecture (using go's naming rather than that of `arch(1)`):

```sh
$ make ARCH=foo ...
```